src/ray/core_worker/core_worker.cc

```diff
@@ -1554,6 +1562,13 @@ Status CoreWorker::CreateActor(const RayFunction &function,
       actor_creation_options.override_environment_variables;
   override_environment_variables.insert(current_override_environment_variables.begin(),
                                         current_override_environment_variables.end());
+  // Update runtime env with simple dict update.
+  ray::RuntimeEnv runtime_env = worker_context_.GetCurrentRuntimeEnv();
+  ray::RuntimeEnv new_runtime_env = actor_creation_options.runtime_env;
+  runtime_env.Update(new_runtime_env);
+  if (runtime_env.conda_env_name == "") {
+    runtime_env = ray::RuntimeEnv::FromProto(job_config_->runtime_env());
+  }
```

Comment:
You're right, this only makes sense because in this PR the runtime env solely consists of the conda name.  I was thinking eventually there could be a flag `runtime_env.empty` that can be used to check if all the fields of `RuntimeEnv` are empty/default. Let me go ahead and add that to this PR.